include/ylt/coro_io/rate_limiter.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2023, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <async_simple/coro/Lazy.h>
+#include <async_simple/coro/SpinLock.h>
+#include <async_simple/coro/SyncAwait.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <ylt/coro_io/coro_io.hpp>
+#include <ylt/easylog.hpp>
+
+namespace coro_io {
+class rate_limiter {
+ public:
+  async_simple::coro::Lazy<std::chrono::milliseconds> acquire(int permits) {
+    co_await this->lock_.coLock();
```

Comment:
use coScopedLock instead of coLock/unlock, see: https://alibaba.github.io/async_simple/docs.en/Lock.html